bpf/tests/bpftest/helpers.go

```diff
@@ -316,6 +323,40 @@ func (tt *Test) Check(t *testing.T, dumpCtx bool) {
 	}
 }
 
+// HasBench returns true if this BPF unit test contains a BENCH program.
+func (tt *Test) HasBench() bool {
+	return tt.benchProg != nil
+}
+
+// Bench runs the PKTGEN, and BENCH programs for a BPF unit test.
+func (tt *Test) Bench() (testing.BenchmarkResult, error) {
+	// create ctx with the max allowed size(4k - head room - tailroom)
+	data := make([]byte, 4096-256-320)
+
+	// ctx is only used for tc programs
+	// non-empty ctx passed to non-tc programs will cause error: invalid argument
+	ctx := make([]byte, 0)
+	if tt.benchProg.Type() == ebpf.SchedCLS {
+		// sizeof(struct __sk_buff) < 256, let's make it 256
+		ctx = make([]byte, 256)
```

Comment:
There's a few magic constants here that seem like they could be better documented through declaration as vars with commetns that reference the corresponding Linux kernel source to explain why these are the right numbers (or hint to someone who has problems how to adjust them).

Reply:
(realizing upon subsequent review these magic constants were already here >.<)